File: Live2DCSharpSDK/Live2DCSharpSDK.WPF/LApp/LAppDefine.cs
using Live2DCSharpSDK.Framework;

namespace Live2DCSharpSDK.WPF.LApp;

/// <summary>
/// 应用程序类。
/// 管理 Cubism SDK。
/// </summary>
public static class LAppDefine
{
    // 屏幕
    public const float ViewScale = 1.0f;
    public const float ViewMaxScale = 2.0f;
    public const float ViewMinScale = 0.8f;

    public const float ViewLogicalLeft = -1.0f;
    public const float ViewLogicalRight = 1.0f;
    public const float ViewLogicalBottom = -1.0f;
    public const float ViewLogicalTop = -1.0f;

    public const float ViewLogicalMaxLeft = -2.0f;
    public const float ViewLogicalMaxRight = 2.0f;
    public const float ViewLogicalMaxBottom = -2.0f;
    public const float ViewLogicalMaxTop = 2.0f;

    // 与外部定义文件(json)保持一致
    public const string MotionGroupIdle = "Idle"; // 待机
    public const string MotionGroupTapBody = "TapBody"; // 点击身体时

    // 与外部定义文件(json)保持一致
    public const string HitAreaNameHead = "Head";
    public const string HitAreaNameBody = "Body";

    // MOC3 一致性校验选项
    public const bool MocConsistencyValidationEnable = true;

    // Framework 输出的日志级别设定
    public const LogLevel CubismLoggingLevel = LogLevel.Verbose;
}

File: web/src/live2d/lappdefine.ts
import { LogLevel } from '@framework/live2dcubismframework';

export const CanvasSize: { width: number; height: number } | 'auto' = 'auto';

export const ViewScale = 1;
export const ViewMaxScale = 2.0;
export const ViewMinScale = 0.2;

export const ViewLogicalLeft = -1.0;
export const ViewLogicalRight = 1.0;
export const ViewLogicalBottom = -1.0;
export const ViewLogicalTop = 1.0;

export const ViewLogicalMaxLeft = -2.0;
export const ViewLogicalMaxRight = 2.0;
export const ViewLogicalMaxBottom = -2.0;
export const ViewLogicalMaxTop = 2.0;

export const ResourcesPath = '/model/';
export const ShaderPath = '/Framework/Shaders/WebGL/';

export const ModelDir: string[] = ['FileReferences_Moc_0'];
export const ModelDirSize: number = ModelDir.length;

export const MotionGroupIdle = 'Tick2';
export const MotionGroupTapBody = '表情组';

export const HitAreaNameDrink = '饮料';
export const HitAreaNameCake = '蛋糕';
export const HitAreaNameBangs = '刘海';
export const HitAreaNameLeftHair = '左侧后发';
export const HitAreaNameRightHair = '右侧后发';

export const PriorityNone = 0;
export const PriorityIdle = 1;
export const PriorityNormal = 2;
export const PriorityForce = 3;

export const MOCConsistencyValidationEnable = true;
export const MotionConsistencyValidationEnable = true;

export const DebugLogEnable = false;
export const DebugTouchLogEnable = false;

export const CubismLoggingLevel: LogLevel = LogLevel.LogLevel_Verbose;
